text/0000-no-bound-ergonomics.md

```diff
@@ -0,0 +1,99 @@
+- Feature Name: auto-bound
+- Start Date: 2018-06-24
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+This document adds a new ergonomics syntax to Rust to express type variables which *bounds* or
+*bounds combination* – are infered from the implementation – in order to *mirror* the already
+existing `impl Trait` in return position that infers the type on the implementation.
+
+```rust
+fn foo(x: &impl _) { // x’s bounds will be deduced from the implementation
+  let y = x.clone(); // we want Clone here
```

Comment:
What happens if there exists a second trait that has a method called `clone`? Why wouldn't that infer no bounds on the type at all, since `&T` is already Clone for any T?